src/main/java/school/lemon/changerequest/java/matrix/Matrix.java

```diff
@@ -0,0 +1,104 @@
+package school.lemon.changerequest.java.matrix;
+
+/**
+ * Created by Yaroslav Pavlinskiy on 12.12.2016.
+ */
+public class Matrix {
+
+    private final int Rows; // count Rows
+    private final int Columns; // count Columns
+    private final int[][] data; // 2d array for matrix
+
+    public Matrix(int R,int C)// Constructor with Rows and Columns
+    {
+        this.Rows = R;
+        this.Columns = C;
+        this.data = new int[R][C];
+    }
+
+    public Matrix(int[][]data) //Constructor 2d array
+    {
+        this.Rows = data.length;
+        this.Columns = data[0].length;
```

Comment:
It's possible to create multidimensional arrays in Java like:
```
        int[][] example = new int[3][];
        example[0] = new int[100];
        example[1] = new int[1000];
        example[2] = new int[0];
```

So determine columns count by the `data[0].length` is not the best approach. Please add verification that all rows has the same columns number. 

By the way, if you will modify this incoming array - your matrix will be changed too. So I'd prefer copying.